pkg/controller/jobs/appwrapper/appwrapper_controller.go

```diff
@@ -120,20 +122,50 @@ func (aw *AppWrapper) GVK() schema.GroupVersionKind {
 }
 
 func (aw *AppWrapper) PodSets() ([]kueue.PodSet, error) {
-	podSets, err := awutils.GetPodSets((*awv1beta2.AppWrapper)(aw))
+	podSpecTemplates, awPodSets, err := awutils.GetComponentPodSpecs((*awv1beta2.AppWrapper)(aw))
 	if err != nil {
-		ctrl.Log.Error(err, "Error returned from awutils.GetPodSets", "appwrapper", aw)
+		ctrl.Log.Error(err, "Error returned from awutils.GetComponentPodSpecs", "appwrapper", aw)
 		return nil, err
 	}
-	for idx := range podSets {
-		podSets[idx].TopologyRequest = jobframework.PodSetTopologyRequest(&podSets[idx].Template.ObjectMeta, nil, nil, nil)
+	podSets := []kueue.PodSet{}
+	for psIndex := range podSpecTemplates {
+		var podIndexLabel *string
+		var subGroupIndexLabel *string
+		var subGroupCount *int32
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASPodIndexLabel]; ok {
+			podIndexLabel = &annotation
+		}
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASSubGroupIndexLabel]; ok {
+			subGroupIndexLabel = &annotation
+		}
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASSubGroupCount]; ok {
+			if count, err := strconv.Atoi(annotation); err == nil {
```

Comment:
Do we want to silently swallow this error?

Reply:
I can add a warning log.